### PR TITLE
Core/Packet: Guild

### PR DESCRIPTION
**Pull request by IntelligentQuantum**
## Changes Proposed:
-  Class base packet
-  cleanup

## Issues Addressed:

- Closes 

## SOURCE:

TC

## Tests Performed:

- SOSO, need test
- 

## How to Test the Changes:

1. Do everything you know about guild like create or delete etc..
2. test guild command
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by claudiodfc** [comment]
## Test results: Positive ✔
## Tests performed:
- .guild create command
- Set MOTD
- Talk in officer chat
- Created new rank
- Invited another player
- Promote and Demote
- Remove created rank
- Created Guild bank
- Bought tabs, deposited gold and items, withdrew as well
- Set guild bank tab information and changed icons and names
- Left guild by non-guild master command
- Guild kicked invited player
- .guild delete command

File: src/server/game/Guilds/Guild.cpp [hunk 1]
```diff
@@ -2407,7 +2330,7 @@ void Guild::_CreateNewBankTab()
     trans->Append(stmt);
 
     ++tabId;
-    for (Ranks::iterator itr = m_ranks.begin(); itr != m_ranks.end(); ++itr)
+    for (auto itr = m_ranks.begin(); itr != m_ranks.end(); ++itr)
         (*itr).CreateMissingTabsIfNeeded(tabId, trans, false);
```
Comment (Winfidonarleyan) [inline src/server/game/Guilds/Guild.cpp:2335]:
```cpp
for (auto const& itr : m_ranks)
```

File: src/server/game/Guilds/Guild.h [hunk 2]
```diff
@@ -661,18 +667,14 @@ class Guild
         void CanStoreItemInTab(Item* pItem, uint8 skipSlotId, bool merge, uint32& count);
     };
 
-    typedef std::unordered_map<ObjectGuid, Member*> Members;
-    typedef std::vector<RankInfo> Ranks;
-    typedef std::vector<BankTab*> BankTabs;
-
 public:
-    static void SendCommandResult(WorldSession* session, GuildCommandType type, GuildCommandError errCode, std::string const& param = "");
+    static void SendCommandResult(WorldSession* session, GuildCommandType type, GuildCommandError errCode, std::string_view param = "");
```
Comment (Winfidonarleyan) [inline src/server/game/Guilds/Guild.h:671]:
```suggestion
    static void SendCommandResult(WorldSession* session, GuildCommandType type, GuildCommandError errCode, std::string_view param = {});
```

File: src/server/game/Guilds/Guild.h [hunk 3]
```diff
@@ -732,17 +734,17 @@ class Guild
...
+    void BroadcastPacket(WorldPacket const* packet) const;
 
     void MassInviteToEvent(WorldSession* session, uint32 minLevel, uint32 maxLevel, uint32 minRank);
 
     template<class Do>
     void BroadcastWorker(Do& _do, Player* except = nullptr)
     {
-        for (Members::iterator itr = m_members.begin(); itr != m_members.end(); ++itr)
-            if (Player* player = itr->second->FindPlayer())
+        for (auto itr = m_members.begin(); itr != m_members.end(); ++itr)
```
Comment (Winfidonarleyan) [inline src/server/game/Guilds/Guild.h:746]:
```cpp
for (auto const& itr : m_members)
```

File: src/server/game/Guilds/Guild.cpp [hunk 4]
```diff
@@ -236,29 +231,31 @@ void Guild::BankEventLogEntry::SaveToDB(CharacterDatabaseTransaction trans) cons
...
+    bankLogEntry.EntryType = int8(m_eventType);
 
     switch(m_eventType)
     {
         case GUILD_BANK_LOG_DEPOSIT_ITEM:
         case GUILD_BANK_LOG_WITHDRAW_ITEM:
-            data << uint32(m_itemOrMoney);
-            data << uint32(m_itemStackCount);
+            bankLogEntry.ItemID = int32(m_itemOrMoney);
+            bankLogEntry.Count = int32(m_itemStackCount);
```
Comment (Winfidonarleyan) [inline src/server/game/Guilds/Guild.cpp:246]: why different types?

Comment (Winfidonarleyan) [reply src/server/game/Guilds/Guild.cpp:246]: Okay, same in TC

File: src/server/game/Guilds/GuildMgr.cpp [hunk 5]
```diff
@@ -63,17 +63,12 @@ Guild* GuildMgr::GetGuildById(uint32 guildId) const
...
-    std::string search = guildName;
-    std::transform(search.begin(), search.end(), search.begin(), ::toupper);
-    for (GuildContainer::const_iterator itr = GuildStore.begin(); itr != GuildStore.end(); ++itr)
-    {
-        std::string gname = itr->second->GetName();
-        std::transform(gname.begin(), gname.end(), gname.begin(), ::toupper);
-        if (search == gname)
-            return itr->second;
-    }
+    for (auto [id, guild] : GuildStore)
```
Comment (Winfidonarleyan) [inline src/server/game/Guilds/GuildMgr.cpp:68]: auto const&

File: src/server/game/Guilds/Guild.h (hunk 3 above)
Comment (IntelligentQuantum) [reply src/server/game/Guilds/Guild.h:746]: can't be const

**Pull request merged**